### PR TITLE
adding redirect hooks

### PR DESCRIPTION
**Pull request by rupadana**
this hooks used to modify socialite scope or other before redirected

```php
public function boot(): void
{
    Socialment::redirect(function(AbstractProvider $provider) {
        return $provider->setScopes([
            'repo','read:user'
        ]);
    });
}
```

### CODE REVIEW

**Comment by chrisreedio** [comment]
@rupadana Can you explain why this is needed?

I'm not seeing a big use case personally.

**Comment by chrisreedio** [comment]
Bringing over some notes from Discord:

```php
->plugins([
    \ChrisReedIO\Socialment\SocialmentPlugin::make()
        ->registerProvider('azure', 'fab-windows', 'Azure'),
        ->registerProvider('github', 'fab-github', 'Github', scopes: ['repo', 'read:user']),
])
```

Something like this would allow for differing scopes per provider, per panel.

Need some additional thoughts for multi-tenancy.

**Comment by rupadana** [comment]
@chrisreedio please review it again

File: src/Http/Controllers/SocialmentController.php [hunk 1]
```diff
@@ -42,8 +42,8 @@ public function redirectPanel(string $provider, string $panelId): RedirectRespon
         if (! request()->session()->exists('socialment.intended.url')) {
             request()->session()->put('socialment.intended.url', $referer);
         }
-
-        return Socialite::driver($provider)->redirect();
+        
+        return Socialment::resolveRedirectCallback($provider, Socialite::driver($provider))->redirect();
```
Comment (chrisreedio) [inline src/Http/Controllers/SocialmentController.php:46]: Would this not need to also be used in the other redirect handlers?

File: src/Socialment.php [hunk 2]
```diff
@@ -4,4 +4,21 @@
 
 class Socialment
 {
+    protected array $providers = [];
+
+    public function getProvider($driver)
+    {
+        return $this->providers[$driver];
```
Comment (chrisreedio) [inline src/Socialment.php:11]: This has been added by another PR in a slightly more enhanced fashion. Please pull and revise.

**Review by chrisreedio — CHANGES_REQUESTED** [review]

**Comment by chrisreedio** [comment]
This has been implemented by #58

**Pull request closed without merging**